### ISSUE TITLE
Long index names can conflict

### ISSUE THREAD

**Issue by blimmer** [joist-orm/joist-orm#1199]
PR #1195 could cause problems with long many-to-many table names. For example:

> CREATE INDEX "change_requests_related_to_change_requests_change_request_related_to_id_index" ON "change_requests_related_to_change_requests" ("change_request_related_to_id");
> error: relation "change_requests_related_to_change_requests_change_request_relat" already exists

Postgres has a maximum index name length of 63. Therefore, people might run into this with our new indexing.

We should write some common logic to maintain uniqueness and avoid character limits.

**Comment by stephenh** [comment]
Makes sense! We should be able to detect "over 63 chars" and rewrite `aaa_bbb_to_ccc_ddd` to `ab_to_cd` or what not. Or maybe keep the 1st part if possible, so `aaa_bbb_to_cd` just to help with sorting the table next to the others.
